modules/apps/info/info-api/src/main/java/com/liferay/info/item/field/reader/InfoItemFieldReaderFieldSetProvider.java

```diff
@@ -25,9 +25,9 @@
  */
 public interface InfoItemFieldReaderFieldSetProvider {
 
-	public InfoFieldSet getInfoFieldSet(String itemClassName);
+	public <T> InfoFieldSet getInfoFieldSet(Class<T> clazz);
```

Comment:
I debated whether to pass a Class or a className but I settled on a className because when this is invoked from a presentation layer, the information available is often a className and retrieving a Class out of it is not obvious.